core/src/main/java/hudson/util/io/RewindableFileOutputStream.java

```diff
@@ -51,8 +54,8 @@ private synchronized OutputStream current() throws IOException {
             if (!closed) {
                 FileUtils.forceMkdir(out.getParentFile());
                 try {
-                    current = new FileOutputStream(out,false);
-                } catch (FileNotFoundException e) {
+                    current = Files.newOutputStream(out.toPath(), StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING);
+                } catch (FileNotFoundException | NoSuchFileException e) {
```

Comment:
It's from `Files.newOutputStream` IIUC

> Checked exception thrown when an attempt is made to access a file that does not exist.

So basically it seems that the JVM had NIH with itself